app/src/androidTest/java/com/github/houseorganizer/houseorganizer/HouseSelectionActivityTestLeave.java

```diff
@@ -0,0 +1,88 @@
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.GrantPermissionRule;
+
+import com.github.houseorganizer.houseorganizer.panels.MainScreenActivity;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+@RunWith(AndroidJUnit4.class)
+public class HouseSelectionActivityTestLeave {
+    private static FirebaseFirestore db;
+    private static FirebaseAuth auth;
+
+    @Rule
+    public GrantPermissionRule permissionRules = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION);
+
+    @BeforeClass
+    public static void createMockFirebase() throws ExecutionException, InterruptedException {
+        FirebaseTestsHelper.startAuthEmulator();
+        FirebaseTestsHelper.startFirestoreEmulator();
+        FirebaseTestsHelper.setUpFirebase();
+
+        db = FirebaseFirestore.getInstance();
+        auth = FirebaseAuth.getInstance();
+        auth.signOut();
+        signInTestUserWithCredentials(TEST_USERS_EMAILS[1], TEST_USERS_PWD[1]);
+    }
+
+    @AfterClass
+    public static void signOut(){
+        auth.signOut();
+    }
+
+    @Rule
+    public ActivityScenarioRule<MainScreenActivity> testRule = new ActivityScenarioRule<>(MainScreenActivity.class);
+
+    @Test
+    public void canLeaveAsMember() throws InterruptedException, ExecutionException {
+        onView(withId(R.id.house_imageButton)).perform(click());
+        onView(withId(R.id.housesView)).perform(RecyclerViewActions.actionOnItemAtPosition(0, RecyclerViewHelper.clickChildViewWithId(R.id.houseName)));
+        onView(withId(R.id.house_imageButton)).perform(click());
+        Thread.sleep(100);
+
+        Map<String, Object> houseData_before = FirebaseTestsHelper.fetchHouseholdData(TEST_HOUSEHOLD_NAMES[0], db);
+        List<String> resident_before = (List<String>) houseData_before.get("residents");
+        Long num_residents_before = (Long) houseData_before.get("num_members");
+
+        onView(withId(R.id.leaveButton)).perform(click());
+        Thread.sleep(100);
+
+        Map<String, Object> houseData_after = FirebaseTestsHelper.fetchHouseholdData(TEST_HOUSEHOLD_NAMES[0], db);
+        List<String> resident_after = (List<String>) houseData_after.get("residents");
+        Long num_residents_after = (Long) houseData_after.get("num_members");
+
+        assertTrue(resident_before.contains(TEST_USERS_EMAILS[1]));
+        assertFalse(resident_after.contains(TEST_USERS_EMAILS[1]));
+        Long expected = num_residents_before - 1;
+        assertEquals(expected, num_residents_after);
+
+        FirebaseTestsHelper.createHouseholds();
+    }
+}
```

Comment:
can this not be moved in HouseSelectionActivityTest ?

Reply:
Sadly not as the device remembers user 1 as owner, it is a strange behaviour but I couldn't make it work